test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartDirectMethodReport.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator.Reports.EdgeHubRestartTest
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    /// <summary>
+    /// This is a counting report to show test result counts, e.g. expect and match counts; and contains a list of unmatched test results.
+    /// </summary>
+    class EdgeHubRestartDirectMethodReport : TestResultReportBase
+    {
+        public EdgeHubRestartDirectMethodReport(
+            string trackingId,
+            string resultType,
+            bool isPassing,
+            long passedMessageCount,
+            Dictionary<string, ulong> messageCount,
```

Comment:
please explain what is stored in this "messageCount" dictionary.

Reply:
So the `messageCount` is the received result from each of the source, namely, the sender and a reciever. this number should be always be equal if the test runs properly.